From: Emil Velikov <emil.velikov@collabora.com>

Flex version 2.6.2 does not expand (define) the yy version of some
function, thus we fail to compile.

Strictly speaking this might be a flex bug, although expanding the few
instances is perfectly trivial and works with 2.6.2 and earlier versions
of flex.

Cc: "12.0 13.0" <mesa-stable@lists.freedesktop.org>
Cc: Mike Lothian <mike@fireburn.co.uk>
Repored-by: Mike Lothian <mike@fireburn.co.uk>
Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
 src/compiler/glsl/glsl_lexer.ll | 6 +++---
 src/mesa/program/program_lexer.l | 6 +++---
 src/compiler/glsl/glcpp/glcpp-lex.l | 2 +-
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/compiler/glsl/glsl_lexer.ll b/src/compiler/glsl/glsl_lexer.ll
index d5e5d4c..e5492bf 100644
--- a/src/compiler/glsl/glsl_lexer.ll
+++ b/src/compiler/glsl/glsl_lexer.ll
@@ -627,12 +627,12 @@ classify_identifier(struct _mesa_glsl_parse_state *state, const char *name)
 void
 _mesa_glsl_lexer_ctor(struct _mesa_glsl_parse_state *state, const char *string)
 {
-   yylex_init_extra(state, & state->scanner);
-   yy_scan_string(string, state->scanner);
+   _mesa_glsl_lexer_lex_init_extra(state, & state->scanner);
+   _mesa_glsl_lexer__scan_string(string, state->scanner);
 }
 
 void
 _mesa_glsl_lexer_dtor(struct _mesa_glsl_parse_state *state)
 {
-   yylex_destroy(state->scanner);
+   _mesa_glsl_lexer_lex_destroy(state->scanner);
 }

diff --git a/src/mesa/program/program_lexer.l b/src/mesa/program/program_lexer.l
index dee66cb..0196635 100644
--- a/src/mesa/program/program_lexer.l
+++ b/src/mesa/program/program_lexer.l
@@ -474,12 +474,12 @@ void
 _mesa_program_lexer_ctor(void **scanner, struct asm_parser_state *state,
 			 const char *string, size_t len)
 {
-   yylex_init_extra(state, scanner);
-   yy_scan_bytes(string, len, *scanner);
+   _mesa_program_lexer_lex_init_extra(state, scanner);
+   _mesa_program_lexer__scan_bytes(string, len, *scanner);
 }
 
 void
 _mesa_program_lexer_dtor(void *scanner)
 {
-   yylex_destroy(scanner);
+   _mesa_program_lexer_lex_destroy(scanner);
 }

diff --git a/src/compiler/glsl/glcpp/glcpp-lex.l b/src/compiler/glsl/glcpp/glcpp-lex.l
index d09441a..41459cd 100644
--- a/src/compiler/glsl/glcpp/glcpp-lex.l
+++ b/src/compiler/glsl/glcpp/glcpp-lex.l
@@ -584,5 +584,5 @@ HEXADECIMAL_INTEGER	0[xX][0-9a-fA-F]+[uU]?
 void
 glcpp_lex_set_source_string(glcpp_parser_t *parser, const char *shader)
 {
-	yy_scan_string(shader, parser->scanner);
+	glcpp__scan_string(shader, parser->scanner);
 }

